### DISCUSSION TITLE
Trying to get the app to work but it doesn't play songs

### DISCUSSION THREAD

**Discussion by RyuuzakiL5** [hellcat707hp/GTRadio#13] [General]
I'm trying to add Independence FM with my own custom songs (yeah, I just like to ride with my car as I were at GTA IV), I like the Gary's comments and weasel news.

So i just extracted the audio files in wav format and litterally copied the structure from here:
https://github.com/hellcat707hp/GTRadio-SampleData/tree/main/Gen3Stations

I'm just making a test, the only song at Songs folder is "billie_jean.mp3". But it never play it. Just ads, news, etc

What am I doing wrong?

Also, I would like to know if its possible to ad folders inside Songs folder, so I could just use my smarphone albuns there without worrying about letting all songs scrambled inside a single folder or just an option to make shortcuts.

**Comment by hellcat707hp** [comment]
Hey, I'm taking a look at this now, and I'll let you know what I find.

**Comment by hellcat707hp** [comment]
@RyuuzakiL5, I tested this out on my phone and emulated device and things seem to be working, so let's hope its just configuration somewhere. 

So, the first thing I noticed was that you utilized the "gen1" station group as your basis, but Independence FM is from GTA4 which is a "gen3" station group game. 
This might be a simple fix, can you post the contents of the `stationGroupInfo.json` file that you have in the `Main folder/station_group_gen1` folder? 

We might be able to just update the `"generation"...` line in that file to be `"generation": 3,`, then force close and reload the app and try again.

**Comment by hellcat707hp** [comment]
@RyuuzakiL5 
> Also, I would like to know if its possible to ad folders inside Songs folder, so I could just use my smarphone albuns there without worrying about letting all songs scrambled inside a single folder or just an option to make shortcuts.

I looked into the shortcuts thing originally because that would be the ideal solution (as that is what the game does), but I don't remember having any luck with it. Part of that is due to Android's permissions which only grant you access to a single folder and its subfolders, not all the folders on the device.
Also, the app doesn't flatten folder structure in that `/Songs` folder either.

**So, yes, for now you do have to manually manage and copy the files into that one folder ☹️**

However, I opened up [a new feature issue](https://github.com/hellcat707hp/GTRadio/issues/14) for better handling this scenario, because I agree this would be a great thing to have work by just pointing it at your existing music library rather than copying files individually. I can't promise when (or if) I'll get to it though, especially with the permissions challenges.
- https://github.com/hellcat707hp/GTRadio/issues/14

**Comment by RyuuzakiL5** [comment]
Ok, I made some edits in the post, so the part of gen1 I manage to fix before the problem with song not being played. So, it seems when you have just one file than the song will never play, after I put more files it started to play.

I'm sorry to post to quick, I don't know if you can see the history of my edits but I manage to fix 2 problems after posting.

Well, excellent project, in fact, if you manage to make this pointing to an existing music library that will be perfect. I know Independence and Self Radio only plays custom music, so the only thing a person needs is a regular player and set songs to random, but, I just like the feeling of a real radio like GTA manage to give us since IV.

**Comment by hellcat707hp** [comment]
Awesome, I'm glad you got it working! 
And thank you for the kind words about the project. I'm glad to see more people than just myself enjoying it.

Hopefully some day I'll be able to make the self-radio stations easier to setup by allowing you to just link to your existing music. If I ever do that, I will be sure to tag you to let you know.
